src/Xablu.WebApiClient/CrossRestApiClient.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using Xablu.WebApiClient.Exceptions;
+
+namespace Xablu.WebApiClient
+{
+    public class CrossRestApiClient
+    {
+        private static Action<RestApiClientOptions> _configureRestApiClient;
+        private static Lazy<IRestApiClient> _restApiClientImplementation = new Lazy<IRestApiClient>(() => CreateRestApiClient(), System.Threading.LazyThreadSafetyMode.PublicationOnly);
+
+        public static bool IsSupported => _restApiClientImplementation.Value == null ? false : true;
+
+        public static void Configure(Action<RestApiClientOptions> options)
+        {
+            _configureRestApiClient = options;
+        }
+
+        public static IRestApiClient Current
+        {
+            get
+            {
+                var ret = _restApiClientImplementation.Value;
+                if (ret == null)
+                {
+                    throw NotImplementedInReferenceAssembly();
+                }
+                return ret;
+            }
+        }
+
+        private static IRestApiClient CreateRestApiClient()
+        {
+#if NETSTANDARD2_0
+			return null;
```

Comment:
From my understanding this would mean that you always have to have a target platform(iOS,Android,UWP) otherwise the lib won't work? Wouldn't it be better to use #IF targetX, targetY, targetZ for all the platforms we support? Or shouldn't we do an #IF netstandard1_1 here?

Reply:
Maybe you are right. The idea is that you always need to have a runtime (be is iOS, Android, .NET Framework or .NET Core) for this library to work. Since .NET Standard is just a specification and not a runtime the library will not work and should not be constructed.